### PR TITLE
doc: integrate the supported releases page

### LINKED ISSUE #353: Add diagram about release branches and elaborate on support policies
We should integrate our supported releases page with more information about what we support.

I think we could get inspiration from this: https://cert-manager.io/docs/installation/supported-releases/#support-policy

### PR DESCRIPTION
**Pull request by gbartolini**
Closed #353

Signed-off-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>

### CODE REVIEW

File: docs/src/supported_releases.md [hunk 1]
```diff
@@ -73,3 +86,54 @@ the CloudNativePG maintainers think is reasonable to support and to test.
 
 !!! Note
     Dates in the future are uncertain and might change. This applies to Kubernetes versions too.
+
+## What we mean by support
+
+Our support window is roughly five months for each release branch (latest two
+minor release, plus an additional month), given that we produce a new final
```
Comment (jsilvela) [inline docs/src/supported_releases.md:93]:
```suggestion
minor releases, plus an additional month), given that we produce a new final
```

File: docs/src/supported_releases.md [hunk 2]
```diff
@@ -73,3 +86,54 @@ the CloudNativePG maintainers think is reasonable to support and to test.
...
+               \  Apr 21, 2022                                      |   last
+                +-------------------+---------------> release-1.15  |   releases
+                                                                    v
+```
+
+We offer two types of support:
+
+Technical support
+:   Technical assistance is offered on a best-effort basis for supported
+    releases only. You can request support from the community on
```
Comment (jsilvela) [inline docs/src/supported_releases.md:122]:
```suggestion
    releases only. You can request support from the community on the
```

File: docs/src/supported_releases.md [hunk 3]
```diff
@@ -73,3 +86,54 @@ the CloudNativePG maintainers think is reasonable to support and to test.
...
+    currently supported releases. Before back-porting a patch, we ask ourselves:
+    *"Does this back-port improve `CloudNativePG`, bearing in mind that we really
+    value stability for already-released versions?"*
+
+
+## Professional support
+
+If you are looking for professional support, please refer to the
+["Support" page in the website](https://cloudnative-pg.io/support/).
+Such vendors might provide service level agreements that include
```
Comment (jsilvela) [inline docs/src/supported_releases.md:137]:
```suggestion
The vendors listed there might provide service level agreements that included
```

File: docs/src/supported_releases.md [hunk 4]
```diff
@@ -64,6 +72,11 @@ Git tags for versions are prepended with `v`.
 The list of supported Kubernetes versions in the above table depends on what
 the CloudNativePG maintainers think is reasonable to support and to test.
 
+At the moment, the CloudNativePG community does not support any additional
+Kubernetes distribution, like Red Hat OpenShift (this might change in the
+future, but it will be addressed by an official policy written by the
```
Comment (jsilvela) [inline docs/src/supported_releases.md:77]:
```suggestion
future, and if it did change, that would result in an official policy written by the
```

File: docs/src/supported_releases.md [hunk 5]
```diff
@@ -73,3 +86,54 @@ the CloudNativePG maintainers think is reasonable to support and to test.
...
+Our support window is roughly five months for each release branch (latest two
+minor release, plus an additional month), given that we produce a new final
+release every two months.
+
+In the below diagram, `release-1.16` is an example of a release branch.
+
+For example, imagining that the latest release is `v1.16.0`, you can expect
+support for both `v1.16.0` and `v1.15.0`.
+
+Only the last patch release of each branch is actually supported.
```
Comment (jsilvela) [inline docs/src/supported_releases.md:101]:
```suggestion
Only the last patch release of each branch is supported.
```

**Review by jsilvela — APPROVED** [review]
Looks good. I have left a few suggestions for consideration.

**Review by mnencia — APPROVED** [review]

**Review by armru — APPROVED** [review]

**Pull request merged**